src/anemoi/training/diagnostics/callbacks/__init__.py

```diff
@@ -1081,58 +86,109 @@ def get_callbacks(config: DictConfig) -> list:  # noqa: C901
                             mode="max",
                             **checkpoint_settings,
                         ),
-                    ],
+                    ]
                 )
             else:
                 LOGGER.debug("Not setting up a checkpoint callback with %s", save_key)
     else:
         # the tensorboard logger + pytorch profiler cause pickling errors when writing checkpoints
         LOGGER.warning("Profiling is enabled - will not write any training or inference model checkpoints!")
+    return None
 
-    if any([config.diagnostics.log.wandb.enabled, config.diagnostics.log.mlflow.enabled]):
-        from pytorch_lightning.callbacks import LearningRateMonitor
 
-        trainer_callbacks.append(
-            LearningRateMonitor(
-                logging_interval="step",
-                log_momentum=False,
-            ),
-        )
+def _get_config_enabled_callbacks(config: DictConfig) -> list[Callback]:
+    """Get callbacks that are enabled in the config as according to CONFIG_ENABLED_CALLBACKS
 
-    if config.diagnostics.eval.enabled:
-        trainer_callbacks.append(RolloutEval(config))
+    Provides backwards compatibility
+    """
+    callbacks = []
+
+    def nestedget(conf: DictConfig, key, default):
+        keys = key.split(".")
+        for k in keys:
+            conf = conf.get(k, default)
+            if not isinstance(conf, DictConfig):
+                break
+        return conf
+
+    for enable_key, callback_list in CONFIG_ENABLED_CALLBACKS:
+        if isinstance(enable_key, list):
+            if not any(nestedget(config, key, False) for key in enable_key):
+                continue
+        elif not nestedget(config, enable_key, False):
+            continue
+
+        if isinstance(callback_list, list):
```

Comment:
I think after the next iteration of the refactor, this part may be obsolete as we don't have lists in the structure above.